### PR TITLE
Adding flowdock & JIRA integration

### PR DESCRIPTION
**Pull request by hanpeter**
Adding listeners for flowdock & JIRA integration. This depends on #16.

### CODE REVIEW

**Comment by hanpeter** [comment]
I am moving away from the JIRA pypi library as the outdated version of pip is limiting what we can use. The library also throws a multiple warning to cron.

File: aws_analysis_tools/ec2_events/jira_listener.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        type=str,
+        help='Password for the Jira user.',
+        default=None,
+    )
+
+
+class JiraListener(object):
+
+    URL_TEMPLATES = {
+        'base': 'https://kruxdigital.jira.com/{url}',
```
Comment (paul-krohn) [inline aws_analysis_tools/ec2_events/jira_listener.py]: There should be a way to at least over-ride this domain-specific URL.

File: aws_analysis_tools/ec2_events/jira_listener.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        if len([c for c in comments if instance_name in c['body']]) < 1:
+            self._logger.debug('Determined issue %s needs a comment', issue['key'])
+
+            start_time = DateTime(event.not_before)
+            end_time = DateTime(event.not_after) if event.not_after is not None else DateTime(9999, 12, 31)
+            body = self.COMMENT_TEMPLATE.format(
+                instance_name=instance_name,
+                # GOTCHA: Change the time to PST for easier calculation
+                start_time=str(start_time.toZone('PST')),
```
Comment (paul-krohn) [inline aws_analysis_tools/ec2_events/jira_listener.py]: Why are we using PST? What happens when we switch to PDT in a few weeks; are the times in the comments off by an hour?

**Comment by paul-krohn** [comment]
:+1: other than the time zone and URL issues.

**Comment by paul-krohn** [comment]
:+1:

**Pull request merged**